File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int ans=0;
     void helper(TreeNode root,int mx)
    {
        if(root==null) return;
        if(root.val>=mx)
            ans++;
        mx=Math.max(mx,root.val);
        helper(root.left,mx);
        helper(root.right,mx);
    }
    public int goodNodes(TreeNode root) {
        helper(root,Integer.MIN_VALUE);
        return ans;
        
    }
}

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    int ans=0;
public:
    void helper(TreeNode* root,int mx)
    {
        if(root==NULL) return;
        if(root->val>=mx)
            ans++;
        mx=max(mx,root->val);
        helper(root->left,mx);
        helper(root->right,mx);
    }
    int goodNodes(TreeNode* root) {
        ans=0;
        helper(root,INT_MIN);
        return ans;
        
    }
};